README.md

```diff
@@ -1,32 +1,504 @@
+                NSString key = [[pair componentsSeparatedByString:@"="] objectAtIndex:0];
+                NSString value = [[pair componentsSeparatedByString:@"="] objectAtIndex:1];
+
+                if ([key isEqualToString:@"event"]) {
+                    [[Swrve sharedInstance] event:value];
+                }
+            }
+            [[Swrve sharedInstance] sendQueuedEvents];
+        }
+        else {
+            [[UIApplication sharedApplication] openURL:url];
+        }
+    });
+}
+```
+
+
+Push Notifications
+-
+
+Push notifications are disabled by default. To enable them, set `SwrveConfig.pushEnabled` to `YES`. In addition, you must tell the Swrve SDK when your app receives a push notification (this is required for push notification reporting in the Swrve service). If you are using iOS 8 Interactive Notifications, you must also declare the custom action categories to the Swrve SDK (see example below).
+
+Your app must also handle the following two situations:
+
+* The app is launched from a push notification.
+* The app receives the push notification while running.
+
+```
+- (BOOL) application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  // Initialize Swrve with custom events that will trigger the push notification dialog
+  SwrveConfig* config = [[SwrveConfig alloc] init];
+  config.pushEnabled = YES;
+  // Let the SDK know about your iOS8 Interactive Notifications if you have any in your app
+  config.pushCategories = [NSSet setWithObjects:YOUR_CATEGORY, nil];
+  [Swrve sharedInstanceWithAppID:your_app_id apiKey:@"your_api_key" config:config launchOptions:launchOptions];
+}
+```
+
+The remainder of the initialization process depends on how you want to request permission from the user to accept push notifications.
+
+### Timing the device token (push permission) request
+
+When a user first installs a push-enabled app, the latest device token for push notifications is requested, which triggers a permission request to the user to accept push notifications from the app. Swrve has built-in functionality to manage the timing of this request—you can time it to one of the following options:
+
+* Request permission at the start of the session. This is the default configuration.
+* Request permission when one of a list of events is triggered.
+
+For more information about uploading device tokens, see [How Do I Upload Push Device Tokens?](http://docs.swrve.com/faqs/push-notifications/upload-push-device-tokens/)
+
+Apple only asks for permission to receive push notifications once per app, under normal circumstances. As a result, when you want to test your push notification integration on a QA device, you must reset the iOS push permissions to test the different timing configurations for push permissions outlined above. For more information, see [How Do I Reset iOS Permissions?](http://docs.swrve.com/faqs/push-notifications/reset-ios-permissions-for-push-notifications/)
```

Comment:
This should probably be the opening to the section about timing the request, and "How Do I Reset iOS Permissions?" is no longer valid for iOS9